### ISSUE TITLE
Numpad Insert key does not switch between Insert and Replace modes in the script editor

### ISSUE THREAD

**Issue by Merlin1846** [godotengine/godot#57826]
### Godot version

3.4.2

### System information

Editor Ubuntu 20.04

### Issue description

Nothing major just that while trying to see if the editor had support for the insert key I noticed that the Numpad insert key does not work regardless of whether or not the num lock is on. Whereas the insert key that is to the right of F12 works fine.

### Steps to reproduce

Just press the key while using the code editor.

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot/issues/38991.

**Comment by EricEzaM** [comment]
What is insert on the keypad? The current godot keys enum does not have such a key.

**Comment by akien-mga** [comment]
At least on X11 none of the keypad function keys seem supported, numlock is basically handled as always on:
```
        { XK_KP_0, Key::KP_0 },
        { XK_KP_1, Key::KP_1 },
        { XK_KP_2, Key::KP_2 },
        { XK_KP_3, Key::KP_3 },
        { XK_KP_4, Key::KP_4 },
        { XK_KP_5, Key::KP_5 },
        { XK_KP_6, Key::KP_6 },
        { XK_KP_7, Key::KP_7 },
        { XK_KP_8, Key::KP_8 },
        { XK_KP_9, Key::KP_9 },
        // same but with numlock
        { XK_KP_Insert, Key::KP_0 },
        { XK_KP_End, Key::KP_1 },
        { XK_KP_Down, Key::KP_2 },
        { XK_KP_Page_Down, Key::KP_3 },
        { XK_KP_Left, Key::KP_4 },
        { XK_KP_Begin, Key::KP_5 },
        { XK_KP_Right, Key::KP_6 },
        { XK_KP_Home, Key::KP_7 },
        { XK_KP_Up, Key::KP_8 },
        { XK_KP_Page_Up, Key::KP_9 },
```

**Comment by EricEzaM** [comment]
Indeed, just tested with https://www.nirsoft.net/utils/keyboard_state_view.html, numpad insert just comes through as `VK_INSERT` on Windows.

Also the easy test - pressing `KP_0` with numlock off does toggle the script editor insert mode in 4.0 master.

**Comment by akien-mga** [comment]
So this might just be a Linux implementation bug then? We could change the `XK_KP_<Function>` mappings to map like their non-`KP` versions (e.g. `XK_Insert` which maps to `Key::INSERT`)

**Comment by EricEzaM** [comment]
I can't test on Linux, but yes, it appears to be relevant to that platform only (although I can't test Mac either)

**Comment by akien-mga** [comment]
Fixed by #66773.
